pkg/apply/apply.go

```diff
@@ -216,6 +218,18 @@ func (t *TopicApplier) applyNewTopic(ctx context.Context) (*NewChangesTracker, e
 	return changes, nil
 }
 
+// helper function to check if there are any actual updates on the topic 
+// when returning applyExistingTopic
+func checkForChanges(updateChanges *UpdateChangesTracker) (*UpdateChangesTracker) {
+	if (updateChanges.NumPartitions == nil &&
+		updateChanges.NewConfigEntries == nil &&
+		updateChanges.UpdatedConfigEntries == nil &&
+		len(updateChanges.MissingKeys) == 0) {
+		return nil
+	}
+	return updateChanges
```

Comment:
This is needed because [here](https://github.com/getsentry/topicctl/pull/14/files#diff-e93827340ebe6b239ff60427dba9f2881d96e7bba9a8a9b07a09cb64582dd73fR243-R248) we initialize the base `UpdateChangesTracker` object for update applies, which will return the empty changes tracker if no changes are made.
This feels dumb and I'm very open to better ways of handling this.